### PR TITLE
Cleanup Filebeat code and structure to make it more readable

### PR DESCRIPTION
**Pull request by ruflin**
* Rename variables to be more meaningful
* Further cleanup separate of log and stding
* Remove references where not needed
* Several small reorganisations

### CODE REVIEW

File: filebeat/crawler/prospector_log.go [hunk 1]
```diff
@@ -225,6 +224,7 @@ func (p *ProspectorLog) checkExistingFile(h *harvester.Harvester, newFile *input
 
 			// Forget about the previous harvester and let it continue on the old file - so start a new channel to use with the new harvester
 			h.Stat.Ignore()
+			//h.SetOffset(0)
```
Comment (urso) [inline filebeat/crawler/prospector_log.go]: what about this line? New but commented out?

File: filebeat/crawler/registrar.go [hunk 2]
```diff
@@ -15,31 +15,28 @@ import (
...
+	stateMutex   sync.Mutex
 }
 
 func NewRegistrar(registryFile string) (*Registrar, error) {
 
 	r := &Registrar{
 		registryFile: registryFile,
 		done:         make(chan struct{}),
+		state:        map[string]FileState{},
+		Channel:      make(chan []*FileEvent, 1),
```
Comment (urso) [inline filebeat/crawler/registrar.go]: +1

File: filebeat/harvester/log.go [hunk 3]
```diff
@@ -255,3 +226,21 @@ func (h *Harvester) updateOffset(increment int64) {
 func (h *Harvester) UpdateState() {
 	h.sendEvent(h.createEvent())
 }
+
+func (h *Harvester) close() {
+	// On completion, push offset so we can continue where we left off if we relaunch on the same file
+	if h.Stat != nil {
+		h.Stat.Offset <- h.getOffset()
```
Comment (urso) [inline filebeat/harvester/log.go]: how about giving this a name? Something like: `h.Stat.Finalized(h.offset)` or `h.Stat.Over(...)`. `Done`, `Finishing`, ...

**Comment by urso** [comment]
LGTM

File: filebeat/harvester/log.go (hunk 3 above)
Comment (ruflin) [inline filebeat/harvester/log.go]: Long term h.Stat will potentially completely disappear, so will this part of the code. The offset should be sent with every even so no special offset reporting will be needed.

File: filebeat/crawler/prospector_log.go (hunk 1 above)
Comment (ruflin) [inline filebeat/crawler/prospector_log.go]: Should be removed, ok to do it in a second PR so no complete rebuild is needed?

**Pull request merged**